addons/html_editor/static/src/core/clipboard_plugin.js

```diff
@@ -181,38 +195,512 @@ export class ClipboardPlugin extends Plugin {
+                // this maps them to the style attribute.
+                if (node.hasAttribute("color") && !node.style["color"]) {
+                    node.style["color"] = node.getAttribute("color");
+                }
+                if (node.hasAttribute("size") && !node.style["font-size"]) {
+                    // FONT size uses non-standard numeric values.
+                    node.style["font-size"] = +node.getAttribute("size") + 10 + "pt";
+                }
+            } else if (
+                ["S", "U"].includes(node.nodeName) &&
+                node.childNodes.length === 1 &&
+                node.firstChild.nodeName === "FONT"
+            ) {
+                // S and U tags sometimes contain FONT tags. We prefer the
+                // strike to adopt the style of the text, so we invert them.
+                const fontNode = node.firstChild;
+                node.before(fontNode);
+                node.replaceChildren(...fontNode.childNodes);
+                fontNode.appendChild(node);
+            }
+            // Remove all illegal attributes and classes from the node, then
+            // clean its children.
+            for (const attribute of [...node.attributes]) {
+                // Keep allowed styles on nodes with allowed tags.
+                if (
+                    CLIPBOARD_WHITELISTS.styledTags.includes(node.nodeName) &&
+                    attribute.name === "style"
+                ) {
+                    node.removeAttribute(attribute.name);
+                    if (["SPAN", "FONT"].includes(node.tagName)) {
+                        for (const unwrappedNode of unwrapContents(node)) {
+                            this.cleanForPaste(unwrappedNode);
+                        }
+                    }
+                } else if (!this.isWhitelisted(attribute)) {
+                    node.removeAttribute(attribute.name);
+                }
+            }
+            for (const klass of [...node.classList]) {
+                if (!this.isWhitelisted(klass)) {
+                    node.classList.remove(klass);
+                }
+            }
+            for (const child of [...node.childNodes]) {
+                this.cleanForPaste(child);
+            }
+        }
+    }
+    /**
+     * Return true if the given attribute, class or node is whitelisted for
+     * pasting, false otherwise.
+     *
+     * @private
+     * @param {Attr | string | Node} item
+     * @returns {boolean}
+     */
+    isWhitelisted(item) {
+        if (item instanceof Attr) {
+            return CLIPBOARD_WHITELISTS.attributes.includes(item.name);
+        } else if (typeof item === "string") {
+            return CLIPBOARD_WHITELISTS.classes.some((okClass) =>
+                okClass instanceof RegExp ? okClass.test(item) : okClass === item
+            );
+        } else {
+            return (
+                item.nodeType === Node.TEXT_NODE ||
+                (item.matches && item.matches(CLIPBOARD_WHITELISTS.nodes))
```

Comment:
I wonder if we shouldn't register whitelisted stuff as resources.
Ex: the list plugin registers list related tags as allowed nodes, because we only know how to handle these if such plugin is installed.

Reply:
Me too, I added a comment. I'd have it also as resource but for community to extend it.